src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java

```diff
@@ -167,4 +177,66 @@
                     }
                 }));
     }
+
+    public Mono<Void> updateStatus(String consentId, String consentRequestId, ConsentStatus status) {
+        return Mono.create(monoSink -> dbClient.begin(res -> {
+            if (res.succeeded()) {
+                Transaction transaction = res.result();
+                update(consentRequestId, status, monoSink, transaction, UPDATE_CONSENT_REQUEST_STATUS_QUERY);
+                update(consentId, status, monoSink, transaction, UPDATE_CONSENT_ARTEFACT_STATUS_QUERY);
+            }else {
+                monoSink.error(new RuntimeException("Error connecting to database. "));
+            }
+        }));
+    }
+
+    private void update(String id, ConsentStatus status, MonoSink<Void> monoSink, Transaction transaction,
+                        String updateStatusQuery) {
+        transaction.preparedQuery(updateStatusQuery,
+                Tuple.of(status.toString(),
+                        LocalDateTime.now(),
+                        id),
+                updateHandler -> {
+                    if (updateHandler.failed()) {
+                        transaction.close();
+                        monoSink.error(new Exception("Failed to update status"));
+                    } else {
+                        transaction.commit();
```

Comment:
you should commit when both update is successful and even calling monoSink.success should be done only once.